src/Uno.UI.RemoteControl.Host/IDEChannel/IdeChannelServerProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO.Pipes;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using StreamJsonRpc;
+using Uno.UI.RemoteControl.Messaging.IdeChannel;
+
+namespace Uno.UI.RemoteControl.Host.IdeChannel;
+
+internal class IdeChannelServerProvider : IIdeChannelServerProvider
+{
+	private readonly ILogger _logger;
+	private readonly IConfiguration _configuration;
+	private readonly IServiceProvider _serviceProvider;
+	private readonly Task<IdeChannelServer?> _initializeTask;
+	private NamedPipeServerStream? _pipeServer;
+	private IdeChannelServer? _ideChannelServer;
+	private JsonRpc? _rpcServer;
+
+	public IdeChannelServerProvider(ILogger<IdeChannelServerProvider> logger, IConfiguration configuration, IServiceProvider serviceProvider)
```

Comment:
If we remove the `IServiceProvider` from the `IdeChannelServer` we can also remove it from here.